numpy/_core/tests/test_api.py

```diff
@@ -406,6 +406,37 @@ def test_copyto():
     # 'dst' must be an array
     assert_raises(TypeError, np.copyto, [1, 2, 3], [2, 3, 4])
 
+
+def test_copyto_cast_safety():
+    with pytest.raises(TypeError):
+        np.copyto(np.arange(3), 3., casting="safe")
+
+    # Can put integer and float scalars safely (and equiv):
+    np.copyto(np.arange(3), 3, casting="equiv")
+    np.copyto(np.arange(3.), 3., casting="equiv")
+    # And also with less precision safely:
+    np.copyto(np.arange(3, dtype="uint8"), 3, casting="safe")
+    np.copyto(np.arange(3., dtype="float32"), 3., casting="safe")
+
+    # But not equiv:
+    with pytest.raises(TypeError):
+        np.copyto(np.arange(3, dtype="uint8"), 3, casting="equiv")
+
+    with pytest.raises(TypeError):
+        np.copyto(np.arange(3., dtype="float32"), 3., casting="equiv")
```

Comment:
I guess it's on the bubble if this should work or not, and indeed almost everyone will want to use safe casting here. NEP 50 doesn't speak to equiv casting at all. I don't have a good intuition if it would be useful for this to error or not though, so I'm not sure if it matters that we're making a specific choice here.

If you think this is the pragmatic choice I'd lean toward trusting you :)

Reply:
I think for equiv all bet's are off.  If anything I would say even Python float to float64 should maybe just not work, but that seemed like the bigger change.
This, in fact, is no change at all.